src/credix_lp/depository.ts

```diff
@@ -0,0 +1,286 @@
+    );
+    const collateralInfo = await collateralToken.getMintInfo();
+    if (!collateralInfo) {
+      throw new Error('Cannot find the collateral mint');
+    }
+    const collateralDecimals = collateralInfo.decimals;
+
+    // Then derive remaining credix internal accounts from other PDAs
+    const credixSigningAuthority = await this.findCredixSigningAuthority(
+      credixGlobalMarketState,
+      credixProgramId
+    );
+    const credixLiquidityCollateral = await this.findCredixLiquidityCollateral(
+      credixSigningAuthority,
+      collateralMint
+    );
+    const credixPass = await this.findCredixPass(
+      credixGlobalMarketState,
+      depository,
+      credixProgramId
+    );
+
+    // Then generate the depository token accounts
+    const depositoryCollateral = await this.findDepositoryCollateralAddress(
+      depository,
+      collateralMint
+    );
+    const depositoryShares = await this.findDepositorySharesAddress(
+      depository,
+      credixSharesMint
+    );
+
+    // Done
+    return new CredixLpDepository(
+      depository,
+      collateralMint,
+      collateralDecimals,
+      collateralSymbol,
+      depositoryCollateral,
+      depositoryShares,
+      credixProgramState,
+      credixGlobalMarketState,
+      credixSigningAuthority,
+      credixLiquidityCollateral,
+      credixSharesMint,
+      credixPass,
+      credixTreasuryCollateral,
+      credixMultisigKey,
+      credixMultisigCollateral,
+      credixProgramId
+    );
+  }
+
+  private static async findDepositoryCollateralAddress(
+    depository: PublicKey,
+    collateralMint: PublicKey
+  ) {
+    return findATAAddrSync(depository, collateralMint)[0];
+  }
+
+  private static async findDepositorySharesAddress(
+    depository: PublicKey,
+    credixSharesMint: PublicKey
+  ) {
+    return findATAAddrSync(depository, credixSharesMint)[0];
+  }
+
+  private static async findCredixProgramState(
+    credixProgramId: PublicKey
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [Buffer.from(CREDIX_LP_INTERNAL_PROGRAM_STATE_NAMESPACE)],
+        credixProgramId
+      )
+    )[0];
+  }
+
+  private static async findCredixGlobalMarketState(
+    credixProgramId: PublicKey
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [Buffer.from(CREDIX_LP_INTERNAL_CREDIX_MARKETPLACE_NAMESPACE)],
+        credixProgramId
+      )
+    )[0];
+  }
+
+  private static async findCredixSigningAuthority(
+    credixGlobalMarketState: PublicKey,
+    credixProgramId: PublicKey
+  ): Promise<PublicKey> {
+    return (
+      await PublicKey.findProgramAddress(
+        [credixGlobalMarketState.toBuffer()],
+        credixProgramId
+      )
+    )[0];
+  }
+
+  private static async findCredixLiquidityCollateral(
+    credixSigningAuthority: PublicKey,
+    collateralMint: PublicKey
+  ): Promise<PublicKey> {
+    return findATAAddrSync(credixSigningAuthority, collateralMint)[0];
```

Comment:
no need for the method to be `async` if it uses `findATAAddrSync`

Reply:
nice catch